src/createHistory.js

```diff
@@ -0,0 +1,22 @@
+import { createStore } from 'redux';
+
+import Actions from './Actions';
+import ActionTypes from './ActionTypes';
+import createHistoryEnhancer from './createHistoryEnhancer';
+import createStoreHistory from './createStoreHistory';
+
+function reducer(state, action) {
+  if (action.type === ActionTypes.LOCATION_UPDATED) {
+    return action.payload;
+  }
+
+  return state;
+}
+
+export default function createHistory(protocol, ...middlewares) {
+  const enhancer = createHistoryEnhancer(protocol, ...middlewares);
+  const store = createStore(reducer, enhancer);
+  store.dispatch(Actions.init());
+
+  return createStoreHistory(store, state => state);
```

Comment:
- `createHistoryEnhancer`: Creates a store enhancer that gives it history-like functionality
- `createStoreHistory`: Wraps a Redux store that has been enhanced with `createHistoryEnhancer` in an API that's compatible with history v2, which allows drop-in use with React Router v2; otherwise, users could just dispatch actions
- `createHistory`: Given a protocol and 0 or more enhancer middlewares, create a `history` object, and otherwise not expose Redux to the user